src/ReaderThemeLoader.php

```diff
@@ -0,0 +1,172 @@
+<?php
+/**
+ * Class ReaderThemeLoader.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP;
+
+use AmpProject\AmpWP\Infrastructure\Conditional;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use AMP_Options_Manager;
+use AMP_Theme_Support;
+use WP_Theme;
+use WP_Customize_Manager;
+use AMP_Reader_Themes;
+
+/**
+ * Switches to the designated Reader theme when template mode enabled and when requesting an AMP page.
+ *
+ * @package AmpProject\AmpWP
+ */
+final class ReaderThemeLoader implements Service, Registerable, Conditional {
+
+	/**
+	 * Reader theme.
+	 *
+	 * @var WP_Theme
+	 */
+	private $theme;
+
+	/**
+	 * Is Reader mode with a Reader theme selected.
+	 *
+	 * @return bool Whether new Reader mode.
+	 */
+	public static function is_needed() {
+		return (
+			AMP_Theme_Support::READER_MODE_SLUG === AMP_Options_Manager::get_option( Option::THEME_SUPPORT )
+			&&
+			AMP_Reader_Themes::DEFAULT_READER_THEME !== AMP_Options_Manager::get_option( Option::READER_THEME )
+		);
+	}
+
+	/**
+	 * Register the service with the system.
+	 *
+	 * @return void
+	 */
+	public function register() {
+		// The following needs to run at plugins_loaded because that is when _wp_customize_include runs. Otherwise, the
+		// most logical action would be setup_theme.
+		add_action( 'plugins_loaded', [ $this, 'override_theme' ], 9 );
```

Comment:
This should use the `Delayed` interface instead, and then `override_theme` would be the new `register`.

Reply:
Not quite, as it has to happen specifically at priority `9`. The `Delayed` interface only provides for the action name generally, so that would mean priority `10`.